website/docs/developer/development-guide/code/best-practices.mdx

```diff
@@ -0,0 +1,72 @@
+---
+id: best-practices
+title: Best Practices
+description: Best practices for coding in ORCA.
+---
+
+import useBaseUrl from '@docusaurus/useBaseUrl';
+
+## Unit Testing
+All code should reach minimum 80% coverage through [Unit Tests](unit-tests.md).
+
+## Code Style
+We use the [Google Style Guide](https://google.github.io/styleguide/pyguide.html) for style elements such as documentation, titling, and structure.
+
+## Stop on Failure
+Failures within ORCA break through to the Cumulus workflow they are a part of.
+To this end, raising an error is preferred over catching the error and returning a null value or error message.
+
+```python
+try:
+    value = function(param)
+except requests_db.DatabaseError as err:
+    logging.error(err)
+    raise
+```
+```python
+if not success:
+    logging.error(f"You may log additional information if desired. "
+                  f"param: {param}")
+    raise DescriptiveErrorType(f'Error message to be raised info Cumulus workflow.'
+```
+Retries can then be configured in the workflow json if desired. See
+[documentation](https://docs.aws.amazon.com/step-functions/latest/dg/concepts-error-handling.html)
+and
+[tutorials](https://aws.amazon.com/getting-started/hands-on/handle-serverless-application-errors-step-functions-lambda/)
+for more information.
```

Comment:
Add some additional verbiage here to talk about what the code below is. Maybe something like:

"The code snippet below shows using a retry with the `copy_to_glacier` lambda in an ingest workflow. You can see by the setting of  `MaxAttempts` it will retry 6 times before failing. `IntervalSeconds` tells us it will retry in 2 seconds. `BackoffRate` tells us that it will add an additional 2 seconds to the wait time before retrying every time it fails."

Reply:
BackoffRate is actually a scalar, but otherwise I got stuff written up.